terraform/diff.go

```diff
@@ -366,6 +375,17 @@ func (d *InstanceDiff) RequiresNew() bool {
 		return false
 	}
 
+	d.mu.Lock()
+	defer d.mu.Unlock()
```

Comment:
`requiresNew` is called from code than already holds the lock. A pattern I often use is have an exported method which takes a mutex and delegates to an private method to do the work. This way locked methods can call one another through the private interface.